File: services/api/app/schemas/analytics.py
from typing import List, Dict
from pydantic import BaseModel

class SentimentCounts(BaseModel):
    positive: int
    negative: int
    neutral: int
    total: int

class SentimentPercentages(BaseModel):
    positive: float
    negative: float
    neutral: float

class WeekDelta(BaseModel):
    positive_delta: float
    negative_delta: float
    neutral_delta: float
    total_delta: int

class AnalyticsSummaryResponse(BaseModel):
    total_tweets: int
    counts: SentimentCounts
    percentages: SentimentPercentages
    average_confidence: float
    timeframe: str
    change_vs_last_week: WeekDelta

class TrendDataPoint(BaseModel):
    timestamp: str
    date: str
    positive: int
    negative: int
    neutral: int
    total: int
    net_sentiment_score: float

class KeywordItem(BaseModel):
    keyword: str
    count: int
    sentiment_dominant: str
    positive_ratio: float

class BrandComparisonItem(BaseModel):
    brand: str
    total_mentions: int
    positive_percentage: float
    negative_percentage: float
    neutral_percentage: float
    net_sentiment_score: float

class BrandComparisonResponse(BaseModel):
    brands: List[BrandComparisonItem]
    generated_at: str

File: packages/shared-types/src/analytics.ts
import { SentimentType } from './sentiment';

export interface SentimentCounts {
  positive: number;
  negative: number;
  neutral: number;
  total: number;
}

export interface SentimentPercentages {
  positive: number;
  negative: number;
  neutral: number;
}

export interface AnalyticsSummary {
  total_tweets: number;
  counts: SentimentCounts;
  percentages: SentimentPercentages;
  average_confidence: number;
  timeframe: string;
  change_vs_last_week: {
    positive_delta: number;
    negative_delta: number;
    neutral_delta: number;
    total_delta: number;
  };
}

export interface SentimentTrendPoint {
  timestamp: string;
  date: string;
  positive: number;
  negative: number;
  neutral: number;
  total: number;
  net_sentiment_score: number; // positive % minus negative %
}

export interface KeywordFrequency {
  keyword: string;
  count: number;
  sentiment_dominant: SentimentType;
  positive_ratio: number;
}

export interface BrandComparisonItem {
  brand: string;
  total_mentions: number;
  positive_percentage: number;
  negative_percentage: number;
  neutral_percentage: number;
  net_sentiment_score: number;
}

export interface BrandComparisonResponse {
  brands: BrandComparisonItem[];
  generated_at: string;
}
